### PR TITLE
Serial flasher

### PR DESCRIPTION
**Pull request by Chleba**
BFW-1815, BFW-1812

### CODE REVIEW

File: src/gui/screen_menu_esp_update.cpp [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+#include "dbg.h"
+#include "string.h"
+
+#include "cmsis_os.h"
+#include "stm32f4xx_hal.h"
+#include "stm32_port.h"
+#include "esp_loader.h"
+// #include "example_common.h"
+#include "esp_flasher.h"
+#define HIGHER_BAUDRATE 115520
```
Comment (DRracer) [inline src/gui/screen_menu_esp_update.cpp:20]: Isn't this a typo? The standard baudrate is 115200

File: src/gui/screen_menu_esp_update.cpp [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+    char tmp_str1[] = "- ESP not connected\n";
+    WriteStr(tmp_str1);
+}
+void ESP_FLASHER::Connect() {
+    char tmp_str[ESP_DESC_SIZE];
+    if (connect_to_target() == ESP_LOADER_SUCCESS) {
+        // msg = Msg::ConnectOK;
+        esp_connected = true;
+        char tmp_str1[] = "- ESP Connected\n";
+        strcpy(tmp_str, tmp_str1);
```
Comment (DRracer) [inline src/gui/screen_menu_esp_update.cpp:60]: Why do you copy the string? Why don't you just swap pointers around?

File: src/gui/screen_menu_esp_update.cpp [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+        i++;
+        d = str[i];
+    }
+    return lines;
+}
+void ESP_FLASHER::WriteStr(char *str) {
+    memset(plan_str, 0, ESP_DESC_SIZE);
+    strcpy(plan_str, str);
+
+    /*     char tmp_str1[ESP_DESC_SIZE]; */
```
Comment (DRracer) [inline src/gui/screen_menu_esp_update.cpp:132]: Does this commented code have some meaning (e.g. for future improvements)? If not it should be removed from the PR

File: src/main.cpp [hunk 4]
```diff
@@ -190,13 +190,15 @@ char uart6slave_line[32];
 static volatile uint32_t minda_falling_edges = 0;
 uint32_t get_Z_probe_endstop_hits() { return minda_falling_edges; }
 
+static uint8_t is_flashing = 0;
```
Comment (DRracer) [inline src/main.cpp:193]: I assume this is a very brutal piece of development code ... is_flashing variable should be probably hidden somewhere in a more appropriate place related to the code being used in.

**Review by DRracer — CHANGES_REQUESTED** [review]
- lib/WUI/resources/fsdata_wui_local.c disappeared and a .c.bak is being added - looks like an unintended accidental change
- as a prototype this PR is probably enough, my main complaint is about the flashing being done in the UI thread. This cannot be merged into the FW in its current state.

File: src/gui/screen_menu_esp_update.cpp (hunk 1 above)
Comment (Chleba) [reply src/gui/screen_menu_esp_update.cpp:20]: It's not used any more. It will be deleted.

**Comment by marekmosna-prusa3d** [comment]
development moved due to @Chleba relocation

**Pull request closed without merging**